pygfx/renderers/wgpu/meshrender.py

```diff
@@ -36,23 +36,24 @@ def mesh_renderer(wobject, render_info):
     if index_buffer:
         n = len(index_buffer.data)
     else:
-        n = len(vertex_buffers[0].data)
+        n = len(geometry.positions.data)
 
     # Collect vertex buffers
-    vertex_buffers = []
-    vertex_buffers.append(geometry.positions)
+    vertex_buffers = {}
+    vertex_buffers[0] = geometry.positions
     if getattr(geometry, "texcoords", None) is not None:
-        vertex_buffers.append(geometry.texcoords)
-
-    # todo: fail if no texcoords are given, make vertex_buffers a dict!
+        vertex_buffers[1] = geometry.texcoords
+    else:
+        # meh, we need to specify a buffer if the shader uses it
+        vertex_buffers[1] = geometry.positions
 
-    # Normals
+    # Normals. Usually it'd be given. If not, we'll calculate it from the vertices.
     if getattr(geometry, "normals", None) is not None:
         normal_buffer = geometry.normals
     else:
-        normal_data = _calculate_normals(geometry.positions.data, index_buffer.data)
+        normal_data = normals_from_vertices(geometry.positions.data, index_buffer.data)
```

Comment:
Can we add an option to toggle if this should be done automatically or if an error should be raised? Depending on the application it might be preferable to have this kind of "validation" instead of auto correcting behaviour :) otherwise this is great!